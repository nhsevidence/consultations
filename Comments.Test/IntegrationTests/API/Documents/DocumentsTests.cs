using Comments.Test.Infrastructure;
using Shouldly;
using System;
using System.Threading.Tasks;
using NICE.Feeds.Tests.Infrastructure;
using Xunit;
using TestBase = Comments.Test.Infrastructure.TestBase;

namespace Comments.Test.IntegrationTests.API.Documents
{
    public class DocumentsTests : TestBase
    {
        [Theory]
        [InlineData(0)]
        [InlineData(-1)]
        public void Get_Documents_Feed_Using_Invalid_Consultation_Id_Throws_error(int consultationId)
        {
            //Arrange
            Func<Task> response;

            // Act
            response = async () =>
            {
                await _client.GetAsync($"/consultations/api/Documents?consultationId={consultationId}");
            };

            // Assert
            response.ShouldThrow<ArgumentException>();
        }

        [Fact]
        public async Task Get_Documents_Feed_Returns_Populated_Feed()
        {
            //Arrange (in base constructor)

            // Act
            var response = await _client.GetAsync("/consultations/api/Documents?consultationId=54");
            response.EnsureSuccessStatusCode();
            var responseString = await response.Content.ReadAsStringAsync();

            // Assert
            responseString.ShouldMatchApproved();
        }

        [Fact]
        public async Task Get_Documents_Feed_Fills_In_Blank_Document_Titles()
        {
            //Arrange (in base constructor)
            FeedToUse = Feed.ConsultationCommentsPublishedDetailNoTitle;

            // Act
            var response = await _client.GetAsync("/consultations/api/Documents?consultationId=54");
            response.EnsureSuccessStatusCode();
            var responseString = await response.Content.ReadAsStringAsync();

            // Assert
            responseString.ShouldMatchApproved();
        }

	    [Fact]
	    public async Task Get_Draft_Preview_Documents_Feed_Returns_Populated_Feed()
	    {
		    //Arrange (in base constructor)

		    // Act
		    var response = await _client.GetAsync("/consultations/api/PreviewDraftDocuments?consultationId=113&documentId=1&reference=GID-NG10186");
		    response.EnsureSuccessStatusCode();
		    var responseString = await response.Content.ReadAsStringAsync();

		    // Assert
		    responseString.ShouldMatchApproved();
	    }

	    [Fact]
	    public async Task Get_Published_Preview_Documents_Feed_Returns_Populated_Feed()
	    {
		    //Arrange (in base constructor)

		    // Act
		    var response = await _client.GetAsync("/consultations/api/PreviewPublishedDocuments?consultationId=113&documentId=1");
		    response.EnsureSuccessStatusCode();
		    var responseString = await response.Content.ReadAsStringAsync();

		    // Assert
		    responseString.ShouldMatchApproved();
	    }
	}
}
