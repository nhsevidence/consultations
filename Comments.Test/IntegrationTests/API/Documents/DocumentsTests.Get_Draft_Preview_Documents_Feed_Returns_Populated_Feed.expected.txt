[
    {
        "consultationId":1,
        "documentId":1,
        "convertedDocument":true,
        "title":"Unstable angina and NSTEMI",
        "chapters":[
            {
                "slug":"introduction",
                "title":"Introduction"
            },
            {
                "slug":"patient-centred-care",
                "title":"Patient-centred care"
            },
            {
                "slug":"key-priorities-for-implementation",
                "title":"Key priorities for implementation"
            },
            {
                "slug":"guidance",
                "title":"1 Guidance"
            },
            {
                "slug":"notes-on-the-scope-of-the-guidance",
                "title":"2 Notes on the scope of the guidance"
            },
            {
                "slug":"implementation",
                "title":"3 Implementation"
            },
            {
                "slug":"research-recommendations",
                "title":"4 Research recommendations"
            },
            {
                "slug":"other-versions-of-this-guideline",
                "title":"5 Other versions of this guideline"
            },
            {
                "slug":"related-nice-guidance",
                "title":"6 Related NICE guidance"
            },
            {
                "slug":"updating-the-guideline",
                "title":"7 Updating the guideline"
            },
            {
                "slug":"appendix-a-the-guideline-development-group-and-nice-project-team",
                "title":"Appendix A: The Guideline Development Group and NICE project team"
            },
            {
                "slug":"appendix-b-the-guideline-review-panel",
                "title":"Appendix B: The Guideline Review Panel"
            },
            {
                "slug":"appendix-c-the-algorithm",
                "title":"Appendix C: The algorithm"
            },
            {
                "slug":"changes-after-publication",
                "title":"Changes after publication"
            },
            {
                "slug":"about-this-guideline",
                "title":"About this guideline"
            }
        ],
        "href":null,
        "sourceURI":"consultations://./consultation/1/document/1"
    },
    {
        "consultationId":1,
        "documentId":2,
        "convertedDocument":true,
        "title":"Sinusitis (acute): antimicrobial prescribing",
        "chapters":[
            {
                "slug":"overview",
                "title":"Overview"
            },
            {
                "slug":"background",
                "title":"Background"
            },
            {
                "slug":"recommendations",
                "title":"Recommendations"
            },
            {
                "slug":"symptoms-and-signs",
                "title":"Symptoms and signs"
            },
            {
                "slug":"summary-of-the-evidence",
                "title":"Summary of the evidence"
            },
            {
                "slug":"other-considerations",
                "title":"Other considerations"
            }
        ],
        "href":null,
        "sourceURI":"consultations://./consultation/1/document/2"
    }
]